akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -264,48 +314,93 @@ class ClusterReceptionistExtension(system: ExtendedActorSystem) extends Extensio
     if (isTerminated)
       system.deadLetters
     else {
-      val numberOfContacts: Int = config.getInt("number-of-contacts")
-      val responseTunnelReceiveTimeout =
-        config.getDuration("response-tunnel-receive-timeout", MILLISECONDS).millis
       val name = config.getString("name")
       // important to use val mediator here to activate it outside of ClusterReceptionist constructor
       val mediator = pubSubMediator
-      system.actorOf(ClusterReceptionist.props(mediator, role, numberOfContacts,
-        responseTunnelReceiveTimeout), name)
+      system.actorOf(ClusterReceptionist.props(mediator, ClusterReceptionistSettings(config)), name)
     }
   }
 }
 
-object ClusterReceptionist {
+object ClusterReceptionistSettings {
+  /**
+   * Create settings from the default configuration
+   * `akka.cluster.client.receptionist`.
+   */
+  def apply(system: ActorSystem): ClusterReceptionistSettings =
+    apply(system.settings.config.getConfig("akka.cluster.client.receptionist"))
 
   /**
-   * Scala API: Factory method for `ClusterReceptionist` [[akka.actor.Props]].
+   * Create settings from a configuration with the same layout as
+   * the default configuration `akka.cluster.client.receptionist`.
    */
-  def props(
-    pubSubMediator: ActorRef,
-    role: Option[String],
-    numberOfContacts: Int = 3,
-    responseTunnelReceiveTimeout: FiniteDuration = 30.seconds): Props =
-    Props(classOf[ClusterReceptionist], pubSubMediator, role, numberOfContacts, responseTunnelReceiveTimeout)
+  def apply(config: Config): ClusterReceptionistSettings =
+    new ClusterReceptionistSettings(
+      role = roleOption(config.getString("role")),
+      numberOfContacts = config.getInt("number-of-contacts"),
+      responseTunnelReceiveTimeout = config.getDuration("response-tunnel-receive-timeout", MILLISECONDS).millis)
 
   /**
-   * Java API: Factory method for `ClusterReceptionist` [[akka.actor.Props]].
+   * Java API: Create settings from the default configuration
+   * `akka.cluster.client.receptionist`.
    */
-  def props(
-    pubSubMediator: ActorRef,
-    role: String,
-    numberOfContacts: Int,
-    responseTunnelReceiveTimeout: FiniteDuration): Props =
-    props(pubSubMediator, Internal.roleOption(role), numberOfContacts, responseTunnelReceiveTimeout)
+  def create(system: ActorSystem): ClusterReceptionistSettings = apply(system)
 
   /**
-   * Java API: Factory method for `ClusterReceptionist` [[akka.actor.Props]]
-   * with default values.
+   * Java API: Create settings from a configuration with the same layout as
+   * the default configuration `akka.cluster.client.receptionist`.
+   */
+  def create(config: Config): ClusterReceptionistSettings = apply(config)
+
+  /**
+   * INTERNAL API
+   */
+  private[akka] def roleOption(role: String): Option[String] = role match {
+    case null | "" ⇒ None
```

Comment:
Option(null) gives None,  so the below line can be Some(role), and you only need to check for "".